src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -324,5 +322,102 @@ public static bool IsKeyValuePair(this Type type, Type? keyValuePairType = null)
 
             return defaultValue;
         }
+
+        /// <summary>
+        /// Returns a list containing the class hierarchy for the given type,
+        /// starting from the current type up to the least derived base type.
+        /// </summary>
+        public static Type[] GetSortedClassHierarchy(this Type type)
+        {
+            var results = new List<Type>();
```

Comment:
>var results = new List<Type>(); [](http://example.com/codeflow?start=12&length=31)

do we care about optimization here? I am asking as we can have a check like 

if (type == typeof(object))
   return Array.Empty<Type>()

Reply:
We could, however `typeof(object)` uses a specialized converter that would never call into this method.